src/main/scala/org/broadinstitute/dsde/workbench/sam/model/SamModel.scala

```diff
@@ -58,20 +58,19 @@ object SamResourceActions {
 }
 
 object SamResourceActionPatterns {
-  val readPolicies = ResourceActionPattern("read_policies")
-  val alterPolicies = ResourceActionPattern("alter_policies")
-  val delete = ResourceActionPattern("delete")
+  val readPolicies = ResourceActionPattern("read_policies", "", false)
+  val alterPolicies = ResourceActionPattern("alter_policies", "", false)
+  val delete = ResourceActionPattern("delete", "", false)
   
-  val sharePolicy = ResourceActionPattern("share_policy::.+")
-  val readPolicy = ResourceActionPattern("read_policy::.+")
+  val sharePolicy = ResourceActionPattern("share_policy::.+", "", false)
+  val readPolicy = ResourceActionPattern("read_policy::.+", "", false)
 }
 
 case class UserStatusDetails(userSubjectId: WorkbenchUserId, userEmail: WorkbenchEmail) //for backwards compatibility to old API
 case class UserStatus(userInfo: UserStatusDetails, enabled: Map[String, Boolean])
 
-case class ResourceActionPattern(value: String) extends ValueObject {
-  lazy val regex = value.r
-  def matches(other: ResourceAction) = regex.pattern.matcher(other.value).matches()
+case class ResourceActionPattern(value: String, description: String, authDomainConstrained: Boolean) {
+  def matches(other: ResourceAction) = value.r.pattern.matcher(other.value).matches()
```

Comment:
Is there any benefit to leaving this case class as it was and creating another case class that has the pattern (as an object) plus the 2 new attributes? This came to mind looking at the changes you have for SamResourceActionPatterns. You were forced to put in dummy values which are not pretty.

Reply:
resolved with faces, should move SamResourceActionPatterns to tests as it is not used in production code